pkg/generators/scm_provider.go

```diff
@@ -48,8 +48,8 @@ func (g *SCMProviderGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha
 		return nil, EmptyAppSetGeneratorError
 	}
 
-	if appSetGenerator.SCMProvider == nil {
-		return nil, EmptyAppSetGeneratorError
+	if appSetGenerator.ApplicationSetTerminalGenerator == nil || appSetGenerator.ApplicationSetTerminalGenerator.SCMProvider == nil {
+		return nil, nil
```

Comment:
These checks are inconsistent. Some return `EmptyAppSetGeneratorError` and others return `nil`. I've standardized on `nil` for no particular reason.

These lines should never be hit. As far as I can tell they're just safety checks because the types technically allow this condition.

Reply:
same here:
https://github.com/argoproj-labs/applicationset/pull/404#discussion_r746707169